Clinical Key.js

```diff
@@ -141,116 +141,95 @@ function doWeb(doc, url){
 		search.getTranslators();
 	}
 	else if (contentType == 'journalArticle') {
-		eid = url.split('/');
-		pii = eid.pop().slice(7);
-		
-		/* can use xpath to get pii; guessing the url is faster
-		var eid = ZU.xpathText(doc, "//ul/@data-eid");
-		pii = eid.slice(7);*/
-		
-		if(!/^S(\d{15}X|\d{16})/.test(pii)){
-			throw new Error('PII incorrectly formatted');
-			return;
-		}
-		
+		try {
+			eid = url.split('/');
+			pii = eid.pop().slice(7);
+			if (!/^S(\d{15}X|\d{16})/.test(pii)){
+				throw new Error('PII from url failed. Trying Xpath');
+			}
+			} catch(e) {
+				Zotero.debug(e);
+				eid = ZU.xpathText(doc, "//ul/@data-eid");
+				pii = eid.slice(7);
+				if (!/^S(\d{15}X|\d{16})/.test(pii)){
+					throw new Error('PII from Xpath failed');
+				}
+			}
 		doSearch(pii, doc);
 	}
 	
 }
 
 //Search & Processing based on CrossRef.js translator
 function doSearch(pii, doc){
-	crossRefQuery = 'http://api.crossref.org/works?query=' + pii
+	crossRefQuery = 'http://api.crossref.org/works?query=' + pii;
+	//acceptHeader = {'Accept': 'application/vnd.crossref-api-message+json; version=1.0'}
 	Zotero.Utilities.HTTP.doGet(crossRefQuery, function(responseText) {
```

Comment:
Kind of (but this is largely my fault; sorry). I guess what I'd specifically like feedback on (which is what I'm actually waiting for) is how to deal with the crossref API version control. Ideally it would just be implemented and we'd move on. However, bc the API version control is not working (or I've missed something), should we implement the translator with the knowledge that it may break in six months? Or, should I spend the time to try and write in an error-catching plan? (I hope that makes sense; I'll try to clarify if not). The error catching plan seems like more effort than it's worth (but I could be short sighted). Similarly, we could just update the translator if the API is fixed and/or updated. Perhaps I'm overthinking it.